### ISSUE TITLE
Source Facebook failed during sync with KeyError: 'id'

### ISSUE THREAD

**Issue by marcosmarxm** [airbytehq/airbyte#7822]
## Enviroment
- **Airbyte version**:  cloud
- **Source Connector and version**: Facebook Marketing 0.2.24
- **Destination Connector and version**: BigQuery Denormalized 0.1.9
- **Severity**: Very Low / Low / Medium / **High** / Critical
- **Step where error happened**: Deploy / **Sync job** / Setup new connection / Update connector / Upgrade Airbyte

## Current Behavior
The user is trying to sync data from Facebook Marketing to Bigquery.
The Source failed during the process with error:
```
[34msource[0m - 2021-11-10 15:51:56 ERROR DefaultAirbyteStreamFactory(internalLog):95 - Encountered an exception while reading stream SourceFacebookMarketing
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 106, in read
    internal_config=internal_config,
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 136, in _read_stream
    for record in record_iterator:
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 181, in _read_incremental
    for record_counter, record_data in enumerate(records, start=1):
  File "/airbyte/integration_code/source_facebook_marketing/streams.py", line 334, in read_records
    result = self.wait_for_job(stream_slice["job"], stream_state=stream_state)
  File "/usr/local/lib/python3.7/site-packages/backoff/_sync.py", line 94, in retry
    ret = target(*args, **kwargs)
  File "/airbyte/integration_code/source_facebook_marketing/streams.py", line 395, in wait_for_job
    date_stop=job["date_stop"],
KeyError: 'id'
```
## Logs
Complete logs: [logs-7600-0.txt](https://github.com/airbytehq/airbyte-cloud/files/7515032/logs-7600-0.txt)

## The Cause

After recent changes str.format fails to find keyword `id`. This part was never tested so having at least a unittest for it would be nice.

**Comment by manavkohli** [comment]
Also seeing this

**Comment by keu** [comment]
should be fixed by #7904
